### ISSUE TITLE
Docs: Links are broken on site

### ISSUE THREAD

**Issue by paddatrapper** [LibreTime/libretime#986]
The links on the docs site under tutorials and howtos (for example http://libretime.org/manual/tutorials/upload-tracks ) return 404s. However under `mkdocs serve` they are found correctly. I think it is something to do with how the web-server is configured on the site server.

**Comment by Robbt** [comment]
This is annoying. I'm not 100% familiar with the Mkdocs setup on Libretime.org perhaps @hairmare has a clue. I'm goingto make this a release-blocker to show that I think it's important that we try to fix this. Was there a change in the code that perhaps made this stop working ?

**Comment by Robbt** [comment]
It looks like the easy solution is to list them under pages with the rest of the files which will add them to the left sidebar. https://github.com/mkdocs/mkdocs/issues/710 - Not sure why this became an issue now but that appears to be the quick fix.

**Comment by Robbt** [comment]
It actually looks like Version 1.1 of MkDocs was released around 2 months ago, the previously referenced issue should actually have been resolved in late 2018, so I'm not 100% sure what the problem is but it could be related to https://github.com/mkdocs/mkdocs/issues/1401 - and/or some kind of configuration change that is needed in how we deploy mkdocs after builds.

**Comment by Robbt** [comment]
Ok, I am trying to simply remove the deprecated pages in mkdocs.yml and add nav instead. Need to look more into how we deploy this as @hairmare set it up initially and it has just worked until now. The long fix would be to add all of the tutorials etc into the navigation under the various subheaders.

**Comment by paddatrapper** [comment]
Why I think it is a web-server config issue is because it works fine with the in-built HTTP server used by MkDocs. We could test it by putting Nginx/Apache in front of the generated files and see if it still occurs.

**Comment by paddatrapper** [comment]
It also could be the version of MkDocs we use. How do we install it? If it is a system package in a distro release from before 2 months ago, then we could still be running into mkdocs/mkdocs#710

**Comment by paddatrapper** [comment]
I've just checked - with a clean build on my system (mkdocs version 1.0.4) the files are successfully generated - `build/docs/manual/tutorials/autodj/index.html`. We need to see if these are built correctly and deployed to the server providing `libretime.org`. If they are, then it is a web-server issue, if they are not, then it is the build/deploy pipeline

**Comment by Robbt** [comment]
Libretime.org is hosted on github through a mechanism of auto builds I believe.

**Comment by Robbt** [comment]
So it appears that travisCI is supposed to send changes to https://github.com/LibreTime/LibreTime.github.io but it also appears that this is no longer happening automatically. Because the latest change merged in #995 is not on LibreTime.org - so this is a bigger problem than missing indexes. I'll continue to look into it and see what I can figure out.

**Comment by Robbt** [comment]
So I did some digging. I have access to Travis-ci.org as an owner of LibreTime and I can see that we have a a GITHUB_TOKEN stored but I believe this is a personal access token created by @hairmare upon setup. In theory I could create a new one and try this but it might break other things with our travisCI integration if I do something wrong. I don't think there is a way for the organization to do this - https://stackoverflow.com/questions/31159275/how-do-i-generate-the-github-oauth-token-for-organization-accounts - but I am pretty new to travisCI. 

It looks like the travis/python.sh is where the mkdocs build command is triggered. Unsure why the last time this was built was in January 23rd as nothing on our side has changed that I can see.

Unrelated I also approved the transfer of the mattermost plugin to LibreTime organization as I saw it was pending.

**Comment by Robbt** [comment]
So the only thing I can try is to add a new API key and replace the existing one on the travis-ci dashboard for LibreTime.Here is some discussion for how to set up the key - https://github.com/travis-ci/travis-ci/issues/2457 and  https://docs.travis-ci.com/user/deployment/releases/ but I wanted to get feedback before I went ahead with this as it may break travisCI integration.

**Comment by paddatrapper** [comment]
It seems to be broken anyway, so go ahead

**Comment by hairmare** [comment]
It needs a personal access token and that can't be created on an orga. I originally created https://github.com/LibreTimeDocBot and added a PAT there. I just requested a new password for the bot user since I can't seem to find the old one...

Feel free to replace it with a personal token, if I can resurrect the doc bot user I might change it back again ;)

**Comment by hairmare** [comment]
The original config was done in #19 btw.

**Comment by Robbt** [comment]
Ok I went ahead and created a PAT with my personal account to see if this fixes it. Not sure if we need to do a new PR commit of a doc update to test this or just rebuilding in travis CI database will work. If it does fix it then it probably makes sense to restore DocBot or create LibreTimeDocBot2 if the password is inaccessible. I suspect that there might be something else failing though because I don't see any reason why the PAT would be the problem (but I don't see anything else that might be the problem either).

**Comment by Robbt** [comment]
Ok I found the issue, but I'm not sure the exact fix. 
I noticed that basically we have a php 7.3 script that was the only one set to python = true, but for some reason after #960 was merged it was no longer being built. Our config file hasn't changed but the way that travisCI is parsing it has and I suspect for some reason it is no longer running the mkdocs code. As of that PR the 4th testing environment php 7.3 with python = true was no longer being built and thus the mkdocs script code is never being called. I have pushed a PR to try to explicitly using dpl v1 as our logs and warnings indicate we are using dpl v2 even though the travisCI code say it is still in beta, maybe they are testing it on free tier by making it default w/o explicitly informing people or maybe they are just parsing it but. I did determine that the travisCI.org site is parsing and reporting the various build logs differently, so something changed.

And the warnings related to dpl v2 started showing up on the code where the docs failed to build so I suspect this is the issue. Worth merging and trying.

**Comment by paddatrapper** [comment]
Github has issues at the moment resulting in 500 errors when I try merge. Please push your changes to master as my laptop is currently shut down

Apr 23, 2020 15:10:55 Robb <notifications@github.com>:

**Comment by Robbt** [comment]
Ok, so my declaration of explicit exclusion of python=true and every version of php other than 7.3 worked whereas our previous usage of explicit inclusion quit working due to some change in the travis CI yml parsing engine. Anyways, it worked and our docs are updated.

**Comment by Robbt** [comment]
Looks like our Libretime.org page navigation is now weird though. Opening a PR to revert the previous change and see if this fixes it.

**Comment by Robbt** [comment]
So the issue is somewhere between switching it to nav: which broke a lot of thing and pages: which I reverted and the intial broken aspects re-emerged. The change at least confirms that the issue can be affected with mkdocs.yml but switching to nav simply made every item a submenu with its own home page and was probably more broken than this.

**Comment by Robbt** [comment]
Ok more speculation on this. I am still not sure what the problem is but switching pages to nav makes every page look like a homepage and so I suspect that it could be related to this issue - https://github.com/mkdocs/mkdocs/pull/1921 and I'm beginning to wonder what version of mkdocs we have the script building. It could be whatever version is bundled with trusty because I believe that is the image vagrant is using, so my next fix will look into using a newer image for travisCI.

**Comment by Robbt** [comment]
So I opened up a new PR to try to test whether the version of mkdocs included with trusty could be the issue and using bionic I ran into 
ERROR: mkdocs 1.0.4 requires PyYAML>=3.10, which is not installed.
So it is safe to assume our dependence upon trusty has put us using an older version of mkdocs which could be triggering the problem above. Whereas the fix for hidden pages would require a newer version of mkdocs. Next thing to do work on getting the python tests to work with python 3 and thus we can shed some of this legacy stuff.
